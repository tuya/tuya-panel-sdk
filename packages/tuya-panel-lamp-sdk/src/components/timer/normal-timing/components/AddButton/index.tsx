import React from 'react';
import { TouchableOpacity, Image, StyleSheet } from 'react-native';
import { Utils } from 'tuya-panel-kit';
import Res from '../../res/index';
import { IAddButtonProps } from '../interface';

const {
  RatioUtils: { convertX: cx, isIphoneX },
} = Utils;
const AddButton: React.FC<IAddButtonProps> = ({ handleAdd, addBtnTintColor }) => {
  return (
    <TouchableOpacity style={styles.bottomAddBtn} activeOpacity={0.8} onPress={handleAdd}>
      <Image source={Res.addBorder} style={styles.addBorder} />
      <Image source={Res.addBtn} style={[styles.addBtn, { tintColor: addBtnTintColor }]} />
      <Image source={Res.plus} style={styles.plus} resizeMode="contain" />
    </TouchableOpacity>
  );
};

AddButton.defaultProps = {
  handleAdd() {},
};

const styles = StyleSheet.create({
  addBorder: {
    height: cx(76),
    position: 'absolute',
    width: cx(76),
    zIndex: 3,
  },
  addBtn: {
    height: cx(76),
    position: 'absolute',
    width: cx(76),
  },
  bottomAddBtn: {
    alignItems: 'center',
    bottom: isIphoneX ? cx(64) : cx(34),
    height: cx(76),
    justifyContent: 'center',
    position: 'absolute',
    right: cx(12),
    width: cx(76),
  },
  plus: {
    height: cx(19),
    position: 'absolute',
    top: cx(18),
    width: cx(19),
  },
});

export default AddButton;
