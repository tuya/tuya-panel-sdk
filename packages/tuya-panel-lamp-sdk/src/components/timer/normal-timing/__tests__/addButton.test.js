import React from 'react';
import { shallow } from 'enzyme';
import AddButton from '../components/AddButton';

describe('AddButton components', () => {
  it('basic render', () => {
    const wrapper = shallow(<AddButton addBtnTintColor="red" />);
    expect(wrapper).toMatchSnapshot();
  });
});
