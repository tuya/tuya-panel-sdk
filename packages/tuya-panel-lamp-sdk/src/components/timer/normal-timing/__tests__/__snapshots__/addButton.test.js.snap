// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton components basic render 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "bottom": 128,
      "height": 152,
      "justifyContent": "center",
      "position": "absolute",
      "right": 24,
      "width": 152,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 152,
        "position": "absolute",
        "width": 152,
        "zIndex": 3,
      }
    }
  />
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 152,
          "position": "absolute",
          "width": 152,
        },
        Object {
          "tintColor": "red",
        },
      ]
    }
  />
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 38,
        "position": "absolute",
        "top": 36,
        "width": 38,
      }
    }
  />
</TouchableOpacity>
`;
